workshops/hcload/README.md

```diff
@@ -0,0 +1,465 @@
+
+
+- To run your code, simply run
+
+```bash
+hcload
+```
+
+> Tip: To stop running your code, use CTRL + C
+
+- If you encounter something like the following during this workshop, simply run `source prepEnv.sh` again and you should be good to go.
+
+> bash: hcload: command not found
+
+You just finished Part 1!!
+
+![Minions cheering](https://cloud-pq5lbfiab.vercel.app/0cheer.gif)
+
+## Part 2: Theory
+
+The Hackclub CDN API only accepts an array of URLs and not direct file uploads. This means we can mirror anything already hosted on a URL, but we can't upload files from our device. To get around this, we'll have to
+
+1. Start a file server on our device
+2. Expose the file server to the internet using [ngrok](https://ngrok.com/)
+3. Send the URL of the ngrok instance to the API
+
+![Visual diagram of the process above: 1. A request is sent from the program to the CDN with the array of URLs. 2. The server sends a request to the ngrok instance as per the URL provided. 3. The ngrok instance responds with the file contents 4. The server hosts the files and responds with the hosted file URLs](https://cloud-hrkhc4qna.vercel.app/0cdn-workflow.png)
+
+## Part 3: Programming the library
+### Part 3.1: Setting up the server
+
+Open up `mod.ts`, and let's get started! 
+
+- First, let's create our HTTP server. We'll be using [Oak](https://github.com/oakserver/oak), a server middleware framework for Deno.
+
+```js
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+
+export default async function (): Promise<void> {
+        const app = new Application()
+        app.use(async (context: any) => {
+            context.response.body = "Server running"
+        })
+
+        app.addEventListener("listen", async ({ port }) => {
+            console.log("HTTP server ready")
+        })
+
+        await app.listen({ port: 20685 })
+}
+```
+
+At the moment, the server simply replies with "Server running" when anything is requested. If you trying running the program (simply run hcload), you should see HTTP server ready, and on the top left of your screen "Server running"
+
+- Next, let's ask for a filePath in our library's default function, and serve the file when an HTTP request reaches our server
+
+```js
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+
+export default async function (filePath: string): Promise<void> { // <--- Changed
+    const app = new Application()
+    app.use(async (context: any) => {
+        await send(context, filePath, { root: '/' }) // <--- Changed
+    })
+
+    app.addEventListener("listen", async ({ port }) => {
+        console.log("HTTP server ready")
+    })
+
+    await app.listen({ port: 20685 })
+}
+```
+
+Modify `hcload.ts` to pass a filePath to the library. Currently, you can pass the `LICENSE` file included with the starter code.
```

Comment:
I'm not very clear on this whole `filePath` thing. Why do we pass in `filePath`? What does it mean? And why are we passing in a LICENSE file?